src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java

```diff
@@ -682,5 +689,26 @@ private int getMaxConcurrentPerNodeBasedOnMatchingLabels(
         return maxConcurrentPerNodeLabeledIfMatch;
     }
 
+    private void updatePauseAction(Task task, CauseOfBlockage cause) {
+        if (task instanceof PlaceholderTask) {
+            PlaceholderTask placeholderTask = (PlaceholderTask)task;
+            try {
+                FlowNode flowNode = placeholderTask.getNode();
+                if (cause != null) {
+                    if (flowNode.getAction(PauseAction.class) == null) {
+                        flowNode.addAction(new PauseAction(cause.getShortDescription()));
+                    }
+                } else {
+                    PauseAction.endCurrentPause(flowNode);
```

Comment:
`PauseAction#endCurrentPause` has the following code:

```java
LOGGER.log(Level.FINE, "‘endCurrentPause’ was called for a FlowNode (‘{0}’) that does not have an active pause. ‘endCurrentPause’ may have already been called.", node.getDisplayName());
```

That code should never get called; it indicates a programming error. Putting a breakpoint here, I see this is getting called a lot with your PR.

Reply:
Done